src/net/udp.rs

```diff
@@ -198,13 +208,13 @@ impl UdpSocket {
         self.inner.connect(SockAddr::from(socket_addr)).await
     }
 
-    /// Sends data on the socket to the given address.
+    /// Sends data on the socket to the given address or to previously connected remote peer.
```

Comment:
Rather than modify this function to take an Optional address, please revert the changes to this comment and the argument and create a new function right above it, `send`. This will more closely mirror the Linux man pages for send and sendto.

The Optional address for the sendmsg_zc is good the way you've changed it in this PR.

This PR also highlights that we don't have a regular `sendmsg` for UDP. You don't have to do anything about that but you or I should open an Issue about that.

Reply:
You want `send` to not accept address and `send_to` to always require address, right?

Reply:
Yes. That would mirror what I see from the Linux man page for send and sendto. Does that sound reasonable to you? I haven't missed either but I haven't had to send flags along with any send yet so I'm just looking for what makes the most sense at least in some vein. With all these ways of sending data, there's bound to be opinions.

Reply:
Only make these changes to net/udp.rs. Both the send and send_to in that file can then use the common io::send_to as you've changed it to optionally take an address.

Reply:
It wont be different from `UdpSocket::write`, do we need it at all then? Maybe leave non ZC methods as they are?

Reply:
I think we should support send_to and send and write for udp. Send and write use different uring opcodes and over time, we will have a builder for some or all of these operations to let the user get at more of the optional fields defined by the io-uring's opcodes and this is a good time to introduce send since it takes no address while send_to does. Even before we expose the options through a builder, the io_uring device offers different opcodes and we can let the user of this library decide which they prefer.

Reply:
added `UdpSocket::send<T: BoundedBuf>(&self, buf: T) -> crate::BufResult<usize, T>`